src/photos/ducks/clustering/albums.js

```diff
@@ -50,17 +68,47 @@ export const findAutoAlbums = async () => {
   ])
   const results = await cozyClient.data.query(autoAlbums, {
     selector: { auto: true },
-    sort: [{ name: 'asc' }]
+    sort: [{ name: 'desc' }]
   })
   return results
 }
 
-export const saveClustering = async (clusters, albums) => {
+const addToClusters = async (clusters, albumsToSave) => {
+  let i = 0
+  for (const photos of clusters) {
+    await addAutoAlbumReferences(photos, albumsToSave[i++])
+  }
+}
+
+const recreateClusters = async (clusters, albumsToSave) => {
+  for (const photos of clusters) {
+    await removeAutoAlbumsReferences(photos, albumsToSave)
+    const album = await createAutoAlbum(photos)
+    await addAutoAlbumReferences(photos, album)
+  }
+  await removeAutoAlbums(albumsToSave)
+}
+
+const createClusters = async clusters => {
   for (const photos of clusters) {
-    if (photos && photos.length > 0) {
-      const album = await getOrCreateAutoAlbum(photos, albums)
-      await createReferences(photos, album)
+    const album = await createAutoAlbum(photos)
+    await addAutoAlbumReferences(photos, album)
+  }
+}
+
+export const saveClustering = async (clusters, albumsToSave) => {
+  if (albumsToSave && albumsToSave.length > 0) {
+    if (clusters.length === albumsToSave.length) {
+      // The clustering structure has not changed: only new photos to add
+      await addToClusters(clusters, albumsToSave)
+    } else {
+      // The clustering structure has changed: remove the impacted clusters
+      // and create the new structure
+      await recreateClusters(clusters, albumsToSave)
     }
+  } else {
+    // No cluster exist yet: create them
+    await createClusters(clusters)
```

Comment:
Instead of having `createClusters ` and `recreateClusters `, I suggest having `createClusters` and `removeClusters` to avoid code duplication.